### PR TITLE
📖  Update documentation for max CoreDNS version

### LINKED ISSUE #6527: Update documentation for max CoreDNS version must be updated
Documentation about [CoreDNS max supported version](https://cluster-api.sigs.k8s.io/reference/versions.html#kubeadm-control-plane-provider-kubeadm-control-plane-controller) must be updated according to https://github.com/kubernetes-sigs/cluster-api/pull/6525

A new line for v1.2 (v1beta1) | v1.9.2 must be added on the last table on the paragraph linked above

/kind documentation

### PR DESCRIPTION
**Pull request by Sonali-Rajput**
**What this PR does / why we need it**:
Update documentation for max CoreDNS version, A new line for v1.2 (v1beta1) | v1.9.2 must be added on the last table on the paragraph.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<6527>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6527

### CODE REVIEW

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 | v0.4 (v1alpha4) | v1.8.4                          |
 | v1.0 (v1beta1)  | v1.8.5                          | 
 | v1.1 (v1beta1)  | v1.8.6                          |
-
+| v1.2 (v1beta1)  | v1.9.2                          |
```
Comment (killianmuldoon) [inline docs/book/src/reference/versions.md:122]: Super tiny nit - could you add a space after this line?

**Comment by killianmuldoon** [comment]
/retitle 📖  Update documentation for max CoreDNS version

**Comment by killianmuldoon** [comment]
@Sonali-Rajput looking good! One tiny nit and I think it's ready. Let me know if you're having any issue with the CLA.

Thanks for your first contribution!

**Comment by killianmuldoon** [comment]
@Sonali-Rajput no need to assign anyone manually to approve this, no matter what the Prow bot says :smile: . 

Can you just sign up as an individual? It shouldn't be an issue unless your organization has some policy against it? If not is your org not listed in the search box?

**Comment by Sonali-Rajput** [comment]
> @Sonali-Rajput no need to assign anyone manually to approve this, no matter what the Prow bot says 😄 .
> 
> Can you just sign up as an individual? It shouldn't be an issue unless your organization has some policy against it? If not is your org not listed in the search box?

yes I just did as an individual , thanks for helping me out 😊

**Comment by Sonali-Rajput** [comment]
> /lgtm
> 
> Congrats @Sonali-Rajput on your first contribution! 🥳

Thank you so much ❤️😊

**Comment by fabriziopandini** [comment]
/lgtm
/approve
Thanks for your first PR!

**Comment by fabriziopandini** [comment]
/hold
Please squash commits

**Comment by fabriziopandini** [comment]
Also add 
Fixes #https://github.com/kubernetes-sigs/cluster-api/issues/6527
in the PR description so the issue gets closed automatically when this PR merges

**Comment by Sonali-Rajput** [comment]
> /hold Please squash commits

I tried to squash commits can you please review

**Comment by fabriziopandini** [comment]
> I tried to squash commits can you please review

It seems there are still two commits, you should get down to 1

**Pull request merged**